Tests/SonarScanner.MSBuild.PreProcessor.Test/RoslynAnalyzerProviderTests.cs

```diff
@@ -98,70 +83,44 @@ public void RoslynConfig_NoAssemblies()
-
-        var actualSettings = testSubject.SetupAnalyzer(settings, sonarProperties, rules, language);
-
-        CheckSettingsInvariants(actualSettings);
-        logger.AssertWarningsLogged(0);
-        logger.AssertErrorsLogged(0);
+        RoslynAnalyzerTestContext context = new(TestContext, [["c:\\assembly1.dll"]], sonarProperties);
 
-        CheckRuleset(actualSettings.RulesetPath, rootFolder, language);
-        CheckTestRuleset(actualSettings.DeactivatedRulesetPath, rootFolder, language);
+        context.ExecuteTest();
 
-        actualSettings.AnalyzerPlugins.Should().BeEmpty();
-        var plugins = new List<string>();
-        mockInstaller.AssertExpectedPluginsRequested(plugins);
+        context.AssertAnalyzerSettingsAreCorrect();
+        context.AssertNoWarningsOrErrors();
+        context.AssertRulesetsAreCorrect();
+        context.AssertExpectedAssemblies([]);
+        context.AssertExpectedPluginsRequested([]);
     }
 
     [TestMethod]
     public void RoslynConfig_ValidProfile()
     {
-        var rootFolder = CreateTestFolders();
-        var logger = new TestLogger();
-        var rules = CreateRules();
-        var language = RoslynAnalyzerProvider.CSharpLanguage;
-        var mockInstaller = new MockAnalyzerInstaller
-        {
-            AnalyzerPluginsToReturn = new List<AnalyzerPlugin>
-            {
-                CreateAnalyzerPlugin("c:\\assembly1.dll"),
-                CreateAnalyzerPlugin("d:\\foo\\assembly2.dll")
-            }
-        };
-        var settings = CreateSettings(rootFolder);
-
         var sonarProperties = new ListPropertiesProvider(new Dictionary<string, string>
         {
             // for ruleset
             {"wintellect.analyzerId", "Wintellect.Analyzers" },
             {"wintellect.ruleNamespace", "Wintellect.Analyzers" },
-
             // to fetch assemblies
             {"wintellect.pluginKey", "wintellect"},
             {"wintellect.pluginVersion", "1.13.0"},
             {"wintellect.staticResourceName", "SonarAnalyzer.zip"},
-
             {"sonaranalyzer-cs.analyzerId", "SonarAnalyzer.CSharp"},
             {"sonaranalyzer-cs.ruleNamespace", "SonarAnalyzer.CSharp"},
             {"sonaranalyzer-cs.pluginKey", "csharp"},
             {"sonaranalyzer-cs.staticResourceName", "SonarAnalyzer.zip"},
             {"sonaranalyzer-cs.nuget.packageId", "SonarAnalyzer.CSharp"},
             {"sonaranalyzer-cs.pluginVersion", "1.13.0"},
             {"sonaranalyzer-cs.nuget.packageVersion", "1.13.0"},
-
             // Extra properties - those started sonar.cs should be included, the others ignored
             {"sonar.vb.testPropertyPattern", "foo"},
             {"sonar.cs.testPropertyPattern", "foo"},
             {"sonar.sources", "**/*.*"},
             {"sonar.cs.foo", "bar"}
         });
+        RoslynAnalyzerTestContext context = new(TestContext, [["c:\\assembly1.dll"], ["d:\\foo\\assembly2.dll"]], sonarProperties);
```

Comment:
`\` encoding was a terrible language design choice. Use verbatim string everywhere where you have a path
```suggestion
        RoslynAnalyzerTestContext context = new(TestContext, [[@"c:\assembly1.dll"], [@"d:\foo\assembly2.dll"]], sonarProperties);
```